src/internal/tab-bar.ts

```diff
@@ -0,0 +1,168 @@
+    }
+
+    div {
+      display: flex;
+    }
+  `;
+
+  /**
+   * Get or set the active tab.
+   */
+  get active(): PlaygroundInternalTab | undefined {
+    return this._active;
+  }
+
+  set active(tab: PlaygroundInternalTab | undefined) {
+    /**
+     * Note the active tab can be set either by setting the bar's `active`
+     * property to the tab, or by setting the tab's `active` property to
+     * true. The two become synchronized according to the following flow:
+     *
+     *   bar click/keydown
+     *           |
+     *           v
+     *   bar.active = tab ---> changed? ---> tab.active = true
+     *           ^                                 |
+     *           |                                 v
+     *   bar tabchange listener        changed from false to true?
+     *           ^                                 |
+     *           |                                 |
+     *           +--- tab dispatches tabchange <---+
+     */
+    const oldActive = this._active;
+    if (tab === oldActive) {
+      return;
+    }
+    this._active = tab;
+    if (oldActive !== undefined) {
+      oldActive.active = false;
+    }
+    if (tab !== undefined) {
+      tab.active = true;
+    } else {
+      // Usually the tab itself emits the tabchange event, but we need to handle
+      // the "no active tab" case here.
+      this.dispatchEvent(
+        new CustomEvent<{tab?: PlaygroundInternalTab}>('tabchange', {
+          detail: {tab: undefined},
+          bubbles: true,
+        })
+      );
+    }
+  }
+
+  private _tabs: PlaygroundInternalTab[] = [];
+  private _active: PlaygroundInternalTab | undefined = undefined;
+
+  render() {
+    return html`
+      <div role="tablist">
+        <slot
+          @slotchange=${this._onSlotchange}
+          @click=${this._activateTab}
+          @keydown=${this._onKeydown}
+          @tabchange=${this._activateTab}
+        ></slot>
+      </div>
+    `;
+  }
+
+  private _onSlotchange(event: Event) {
+    this._tabs = (
+      event.target as HTMLSlotElement
+    ).assignedElements() as PlaygroundInternalTab[];
+    let newActive;
+    // Manage the idx and active properties on all tabs. The first tab that
+    // asserts it is active wins.
+    for (let i = 0; i < this._tabs.length; i++) {
+      const tab = this._tabs[i];
+      tab.index = i;
+      if (newActive !== undefined) {
+        tab.active = false;
+      } else if (tab.active || tab.hasAttribute('active')) {
+        // Check both the active property and the active attribute, because the
+        // user could have set the initial active state either way, and it might
+        // not have reflected to the other yet.
+        newActive = tab;
+      }
+    }
+    this.active = newActive;
+  }
+
+  private _activateTab(event: Event) {
+    const tab = this._findEventTab(event);
+    if (tab === undefined || tab === this.active) {
+      return;
+    }
+    this.active = tab;
+    tab?.scrollIntoView({behavior: 'smooth'});
```

Comment:
Might this cause issues on page load? Like scroll to your tabs no matter where you are on the page? How does this affect a docs page with multiple playground IDEs on the page?

Reply:
Oh yeah, good point!

Replaced with a  custom implementation. I think this new approach is also better because it also tries to make sure adjacent tabs are at least 48px visible, otherwise they can disappear so that it's hard to navigate to the next/prev one.